### PR TITLE
Angular UI migration to AdminLTE 3

### PR DESCRIPTION
**Pull request by iyilm4z**
Fixes [this](https://github.com/aspnetboilerplate/aspnetboilerplate/issues/5452) issue. Also related to [this](https://github.com/aspnetboilerplate/aspnetboilerplate/issues/4943).

In order to make the pr easy to review, i made commits for only related changes.
That's why i hope you can merge the pr faster than the [previous](https://github.com/aspnetboilerplate/module-zero-core-template/pull/489) one for mvc ui.

After this pr merged, i'll notify admin-lte community about boilerplate's usage of their template.
This will make each big communities closer to each other and most probably prime contributor of admin-lte would like to add our name to their docs under heading of implementations [here](https://adminlte.io/docs/3.0/implementations.html). He did this for my project named [admin-lte-dotnet](https://github.com/iyilm4z/admin-lte-dotnet).

If you need any help, please let me give you a hand for making your job done quickly.
Bye.

### CODE REVIEW

**Comment by demirmusa** [comment]
Hi @iyilm4z 
Thank you for your contributions. I will review it asap.

**Comment by iyilm4z** [comment]
Hi @demirmusa,
Your welcome.  I had an idea of starting a new open source e-commerce project and i wanted to use admin-lte + boilerplate + angular. Changes in this pr were actually for that project but i wanted to share it with all community. Until you merge the pr, i'll park my development on client side of my project because i want to be on the same page with you.
Also thank you for taking time.

**Comment by iyilm4z** [comment]
@demirmusa thanks for your review. I’m happy to see that most of the migration is ok. This time i made each change-set as minimal as possible. Did it helped to review easier?

I’ll do other changes sooner. Let’s keep in touch and finish this job faster.

**Comment by demirmusa** [comment]
> @demirmusa thanks for your review. I’m happy to see that most of the migration is ok. This time i made each change-set as minimal as possible. Did it help to review easier?

Yes it did. Thanks.

File: angular/src/shared/directives/equal-validator.directive.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export class EqualValidator implements Validator {
     // value not equal
     if (control2 && value !== control2.value && !this.isReverse) {
       return {
-        validateEqual: false
+        validateEqual: true
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/shared/directives/equal-validator.directive.ts:39]: yes, it works in this way

File: angular/src/shared/directives/equal-validator.directive.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export class EqualValidator implements Validator {
 
     // value not equal and reverse
     if (control2 && value !== control2.value && this.isReverse) {
-      control2.setErrors({ validateEqual: false });
+      control2.setErrors({ validateEqual: true });
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/shared/directives/equal-validator.directive.ts:53]: yes, it works in this way

File: angular/src/shared/components/validation/abp-validation.summary.component.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+} from '@angular/core';
+import { AbstractControl } from '@angular/forms';
+import { AppComponentBase } from '@shared/app-component-base';
+import { AbpValidationError } from './abp-validation.api';
+
+@Component({
+  selector: 'abp-validation-summary',
+  templateUrl: './abp-validation.summary.component.html'
+})
+export class ApbValidationSummaryComponent extends AppComponentBase
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/shared/components/validation/abp-validation.summary.component.ts]: fixed

File: angular/src/app/layout/sidebar.component.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { Component, ChangeDetectionStrategy, Renderer2 } from '@angular/core';
+import { LayoutStoreService } from '@shared/layout/layout-store.service';
+
+@Component({
+  selector: 'sidebar',
+  templateUrl: './sidebar.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SidebarComponent {
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/app/layout/sidebar.component.ts]: added

File: angular/src/app/users/reset-password/reset-password.component.html [hunk 5]
```diff
@@ -1,28 +1,68 @@
...
+>
+  <abp-modal-header
+    [title]="'ResetPassword' | localize"
+    (onCloseClick)="bsModalRef.hide()"
+  ></abp-modal-header>
+  <div class="modal-body">
+    <div class="row">
+      <div class="col-md-9 offset-md-3">
+        <p class="text-info mb-1">
+          1. Enter your administrator password
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/app/users/reset-password/reset-password.component.html]: done

File: angular/src/app/users/reset-password/reset-password.component.html [hunk 6]
```diff
@@ -1,28 +1,68 @@
...
+        <abp-validation-summary
+          [control]="adminPasswordModel"
+          [controlEl]="adminPasswordEl"
+        ></abp-validation-summary>
+      </div>
+    </div>
+    <div class="row">
+      <div class="col-md-9 offset-md-3">
+        <p class="text-info mb-1">
+          2. Copy this random password so you can send it to the user
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/app/users/reset-password/reset-password.component.html]: done

File: angular/src/shared/shared.module.ts [hunk 7]
```diff
@@ -8,146 +8,44 @@ import { AppSessionService } from './session/app-session.service';
...
-    MatStepperModule,
-    MatTableModule,
-    MatTabsModule,
-    MatToolbarModule,
-    MatTooltipModule,
-    MatTreeModule,
-} from '@angular/material';
+
+import { AbpPaginationControlsComponent } from './components/pagination/abp-pagination-controls.component';
+import { ApbValidationSummaryComponent } from './components/validation/abp-validation.summary.component';
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/shared/shared.module.ts]: fixed

File: angular/src/shared/shared.module.ts [hunk 8]
```diff
@@ -8,146 +8,44 @@ import { AppSessionService } from './session/app-session.service';
...
+        AbpModalHeaderComponent,
+        AbpModalFooterComponent,
         LocalizePipe,
         BlockDirective,
         BusyDirective,
         EqualValidator
     ],
     exports: [
         AbpPaginationControlsComponent,
+        ApbValidationSummaryComponent,
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/shared/shared.module.ts]: fixed

File: angular/src/shared/shared.module.ts [hunk 9]
```diff
@@ -8,146 +8,44 @@ import { AppSessionService } from './session/app-session.service';
...
-        MatTableModule,
-        MatTabsModule,
-        MatToolbarModule,
-        MatTooltipModule,
-        MatTreeModule,
+        NgxPaginationModule
     ],
     declarations: [
         AbpPaginationControlsComponent,
+        ApbValidationSummaryComponent,
```
[the remark this answers is not in the document]
Comment (iyilm4z) [reply angular/src/shared/shared.module.ts]: fixed

**Comment by iyilm4z** [comment]
Hi @demirmusa, code review fixes done. We should also update screenshots.

**Comment by demirmusa** [comment]
Hi @iyilm4z,
As I see, your pr is merged. Thank you so much for your big contrubitions

> Note: We should also update screenshots. You want me do it as well or you'll do it yourself?

We will handle it 👍 .

**Comment by iyilm4z** [comment]
@demirmusa your welcome👍

**Pull request merged**